### PR TITLE
feat(dunning): Create payment request

### PR DESCRIPTION
**Pull request by rsempe**
## Context

We want to be able to manually request payment of the overdue balance and send emails for reminders.

https://getlago.canny.io/feature-requests/p/send-reminders-for-overdue-invoices

## Description

The goal of this PR is to:

- create the payable group for the overdue invoices
- create the payment request for the payable group
- send the payment_request.created webhook

on the payment request creation service

### CODE REVIEW

File: app/services/payment_requests/create_service.rb [hunk 1]
```diff
@@ -10,19 +10,34 @@ def initialize(organization:, params:)
...
+      return result.not_found_failure!(resource: "invoice") if invoices.empty?
 
-        # TODO: Create payment request for the payable group
+      # TODO: Return error if invoices are not overdue
 
-        # TODO: Send payment_request.created webhook
+      ActiveRecord::Base.transaction do
+        # NOTE: Create payable group for the overdue invoices
+        payable_group = customer.payable_groups.create!(organization:)
+        invoices.each { |i| i.update!(payable_group:) }
```
Comment (ancorcruz) [inline app/services/payment_requests/create_service.rb:23]: would you expect any side effect or validation to happen here. if not we could use `update_all` to perform a unique SQL statement in order to update all the invoices at once.

what do you think?

File: app/services/payment_requests/create_service.rb [hunk 2]
```diff
@@ -10,19 +10,34 @@ def initialize(organization:, params:)
...
+        payment_request = payable_group.payment_requests.create!(
+          organization:,
+          customer:,
+          amount_cents: invoices.sum(:total_amount_cents),
+          amount_currency: invoices.first.currency,
+          email:
+        )
+
+        # NOTE: Send payment_request.created webhook
+        SendWebhookJob.perform_later("payment_request.created", payment_request)
```
Comment (ancorcruz) [inline app/services/payment_requests/create_service.rb:35]: we are not testing the `SendWebhookJob` is queued.

**Review by ancorcruz — APPROVED** [review]
I have left 2 comments but it looks good to me!

File: app/services/payment_requests/create_service.rb (hunk 2 above)
Comment (rsempe) [reply app/services/payment_requests/create_service.rb:35]: Good spot 👍

File: app/services/payment_requests/create_service.rb (hunk 1 above)
Comment (rsempe) [reply app/services/payment_requests/create_service.rb:23]: Yeah - I was hesitant but agreed let's update all the invoices at once 👍

**Pull request merged**